avocado/utils/utils_misc.py

```diff
@@ -0,0 +1,262 @@
+            time.sleep(step)
+
+    if not success:
+        raise OSError(39,
+                      "Could not delete directory %s "
+                      "after %d s and %d attempts." %
+                      (path, timeout, attempts))
+
+
+def umount(src, mount_point, fstype, verbose=False, fstype_mtab=None):
+    """
+    Umount the src mounted in mount_point.
+
+    :src: mount source
+    :mount_point: mount point
+    :type: file system type
+    :param fstype_mtab: file system type in mtab could be different
+    :type fstype_mtab: str
+    """
+    if fstype_mtab is None:
+        fstype_mtab = fstype
+
+    if is_mounted(src, mount_point, fstype, None, verbose, fstype_mtab):
+        umount_cmd = "umount %s" % mount_point
+        try:
+            process.system(umount_cmd, verbose=verbose)
+            return True
+        except process.CmdError:
+            return False
+    else:
+        logging.debug("%s is not mounted under %s", src, mount_point)
+        return True
+
+
+def mount(src, mount_point, fstype, perm=None, verbose=False,
+          fstype_mtab=None):
+    """
+    Mount the src into mount_point of the host.
+
+    :src: mount source
+    :mount_point: mount point
+    :fstype: file system type
+    :perm: mount permission
+    :param fstype_mtab: file system type in mtab could be different
+    :type fstype_mtab: str
+    """
+    if perm is None:
+        perm = "rw"
+    if fstype_mtab is None:
+        fstype_mtab = fstype
+
+    if is_mounted(src, mount_point, fstype, perm, verbose, fstype_mtab):
+        logging.debug("%s is already mounted in %s with %s",
+                      src, mount_point, perm)
+        return True
+    mount_cmd = "mount -t %s %s %s -o %s" % (fstype, src, mount_point, perm)
+    try:
+        process.system(mount_cmd, verbose=verbose)
+    except process.CmdError:
+        return False
+    return is_mounted(src, mount_point, fstype, perm, verbose, fstype_mtab)
+
+
+def is_mounted(src, mount_point, fstype, perm=None, verbose=False,
+               fstype_mtab=None):
+    """
+    Check mount status from /etc/mtab
+
+    :param src: mount source
+    :type src: string
+    :param mount_point: mount point
+    :type mount_point: string
+    :param fstype: file system type
+    :type fstype: string
+    :param perm: mount permission
+    :type perm: string
+    :param verbose: if display mtab content
+    :type verbose: Boolean
+    :param fstype_mtab: file system type in mtab could be different
+    :type fstype_mtab: str
+    :return: if the src is mounted as expect
+    :rtype: Boolean
+    """
+    if perm is None:
+        perm = ""
+    if fstype_mtab is None:
+        fstype_mtab = fstype
+
+    # Version 4 nfs displays 'nfs4' in mtab
+    if fstype == "nfs":
+        fstype_mtab = r"nfs\d?"
+
+    mount_point = os.path.realpath(mount_point)
+    if fstype not in ['nfs', 'smbfs', 'glusterfs', 'hugetlbfs']:
+        if src:
+            src = os.path.realpath(src)
```

Comment:
This would have to be skipped for custom `process.run`, or at least changed to `run("realpath %s" % src)` in order to execute on the right machine.